javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,183 @@
+//Item array removal
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+  ];
+  const nameToRemove = names.indexOf("Ahmad");
+
+  if(nameToRemove>-1){
+      names.splice(nameToRemove,1); //remove one item only
+  };
+  
+  console.log(names); // ['Peter', 'Yana', 'kristina', 'Rasmus', 'Samuel', 'katrine', 'Tala']
+
+
+  //When will we be there? exircise 
+
+
+  const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+  };
+
+  function calculateTime(speed, km){
+const timeInMinutes =(km/speed)*60; // turn the num to minutes
+const hours = Math.floor(timeInMinutes/60); //gets the hours
+const rhours= Math.floor(hours);  // should give us the hours
+const minutes =Math.floor(hours-rhours)*60; //takes the decimal part and multiplate it by 60
+const rminutes= Math.floor(minutes); // should give us the minutes
+console.log (`${rhours} hours and ${rminutes} minutes`);
+return;
+    
+    };  
+
+
+calculateTime(travelInformation.destinationDistance, travelInformation.speed);
+ //not sure why the output is always 0 I tried many ways to solve it..
+  
+
+
+  ///Series duration of my life 
+
+const seriesDurations=[
+  {
+    title: "Gotham",
+    days: 2,
+    hours: 23,
+    minutes: 40,
+  },
+  {
+    title: "The walking dead",
+    days: 7,
+    hours: 23,
+    minutes: 58,
+  },
+  {
+    title: "Rick and Morty",
+    days: 8,
+    hours: 16,
+    minutes: 40,
+  },
+];
+function logOutSeriesText() {
+  let totalSeriesDurations= 0;  // countinually update the variable
+
+  for (let serie of seriesDurations){
+    const convertToMinutes= serie.days*1440 + serie.hours*60 + serie.minutes;
+    const averageYears = 80*525600; // years to minutes
+   totalSeriesDurations += convertToMinutes/averageYears *100 ;  // for each serie duration time Im adding it to total
+   console.log(`${serie.title} took ${totalSeriesDurations.toFixed(3)} % of my life`);
+  }
+  const totalAverage=totalSeriesDurations/ seriesDurations.length;
+  console.log(`In total that is ${totalAverage.toFixed(2)} % of my life`);
+}
+logOutSeriesText(); 
+
```

Comment:
Great work Hanna, you are really close to solving the problem 💫💫
Have a closer look at line 74, you should not be aggregating the values of all the minutes per series in this line, do you see why? In other words, delete the '+' before '=', such that the line is 
totalSeriesDurations = convertToMinutes/averageYears *100;

You do however want an additional variable through which you store the total amount of minutes spent on all series in the array, and then you should take the percentage of this. Try to rewrite line 77 and 78 with this in mind 😊

Lastly, have a closer look at your variable naming again, can you change some of the names to represent the values more accurately?

Reply:
Im a bit confuse about how to add and store the total amount of all the series,  in the loop I get the amount spend in each serie but how do I add them together?